ui/src/pages/distribution/auction/Auction.tsx

```diff
@@ -47,6 +50,34 @@ export const Auction: React.FC<RouteComponentProps & Props> = ({
 
   const allBiddingAuctions = useStreamQueries(BiddingAuction).contracts;
   const { contracts: allBids, loading: allBidsLoading } = useStreamQueries(Bid);
+  const { contracts: deposits, loading: depositsLoading } = useStreamQueries(AssetDeposit);
+
+  const addSignatoryAsDepositObserver = async (
+    deposit: CreateEvent<AssetDeposit>,
+    newObs: string[]
+  ) => {
+    const newObservers = makeDamlSet([...deposit.observers, ...newObs]);
+
+    await ledger.exercise(AssetDeposit.AssetDeposit_SetObservers, deposit.contractId, {
+      newObservers,
+    });
+  };
+
+  const updateDeposits: any = async (retries: number) => {
+    if (retries > 0) {
+      return updateDeposits(retries - 1);
+    }
+    return Promise.all(
+      deposits.map(d => {
+        const newObservers = damlSetValues(d.payload.asset.id.signatories).filter(
+          signatory => !d.observers.includes(signatory)
+        );
+        if (newObservers.length > 0) {
+          addSignatoryAsDepositObserver(d, newObservers);
+        }
+      })
+    );
+  };
```

Comment:
Is this also so the issuer is added as an observer to the `AssetDeposit` ?

Reply:
See my reply above ^